app/models/article.rb

```diff
@@ -35,7 +35,9 @@ def name
   end
 
   def path
-    if published?
+    if published? && short_path
```

Comment:
@veganstraightedge question here, Which URL should take precedence? I'm making a decision right now that short_path takes over if it exists and generates the paths, but i'm unsure this is correct.